src/typechecker.erl

```diff
@@ -89,8 +88,8 @@ subtypes([Ty1|Tys1], [Ty2|Tys2], TEnv) ->
 %% The function compat_ty is just a convenience function to be able to
 %% pattern match on types in a nice way.
 compat(T1, T2, A, TEnv) ->
-    Ty1 = normalize(T1, TEnv),
-    Ty2 = normalize(T2, TEnv),
+    Ty1 = typelib:remove_pos(normalize(T1, TEnv)),
```

Comment:
This makes a lot of sense to me! Good solution.

Reply:
I agree, it's a very nice and surgical fix.